inst/include/common/information.hpp

```diff
@@ -149,6 +157,77 @@ class Information {
     this->random_effects_parameters.push_back(&re);
   }
 
+  /**
+   * Register a parameter name.
+   *
+   * @param p_name
+   */
+  void RegisterParameterName(std::string p_name) {
+    this->parameter_names.push_back(p_name);
+  }
+
+    
+  void setup_priors(){
+    for(density_components_iterator it = density_components.begin(); it!= density_components.end(); ++it){
```

Comment:
Do we need to use this->density_components.begin() and this->density_components.end()?

The three functions setup_priors, setup_random_effects(), and setup_data have very similar structures. Is the primary difference between them the d->input_type? Is it possible to create a generic function to handle the shared logic, then each of the three functions only need to pass in "prior", "random_effects", or "data" as needed?